### PR TITLE
frontend: Add the CreatePassword mutation

### PR DESCRIPTION
**Pull request by ryanslade**
Only for users that initially signed up with OAuth and subsequently
removed all their OAuth connections.

It can only create a new password where the current password 
is null and the user doesn't have any external login connections.

Closes: https://github.com/sourcegraph/sourcegraph/issues/17729

### CODE REVIEW

File: internal/database/users.go [hunk 1]
```diff
@@ -905,6 +905,9 @@ func (u *UserStore) SetPassword(ctx context.Context, id int32, resetCode, newPas
 	if newPassword == "" {
 		return false, errors.New("new password was empty")
 	}
+	if err := CheckPasswordLength(newPassword); err != nil {
```
Comment (ryanslade) [inline internal/database/users.go:908]: This seems like an oversight in the existing code

File: internal/database/users.go [hunk 2]
```diff
@@ -974,6 +977,59 @@ func (u *UserStore) UpdatePassword(ctx context.Context, id int32, oldPassword, n
 	return nil
 }
 
+// CreatePassword creates a user's password iff don't have a password and they
```
Comment (arussellsaw) [inline internal/database/users.go:980]: small typo here
```suggestion
// CreatePassword creates a user's password if they don't have a password and they
```

**Review by arussellsaw — APPROVED** [review]

File: internal/database/users.go (hunk 2 above)
Comment (ryanslade) [reply internal/database/users.go:980]: That was actually intentional 😄 

https://en.wikipedia.org/wiki/If_and_only_if#Origin_of_iff_and_pronunciation

**Review by artemruts — APPROVED** [review]
Awesome, I'll merge this into my branch to finalize UI.

File: internal/database/users.go [hunk 3]
```diff
@@ -974,6 +977,59 @@ func (u *UserStore) UpdatePassword(ctx context.Context, id int32, oldPassword, n
...
+// don't have any valid login connections.
+func (u *UserStore) CreatePassword(ctx context.Context, id int32, password string) error {
+	u.ensureStore()
+
+	// 🚨 SECURITY: Check min and max password length
+	if password == "" {
+		return errors.New("new password was empty")
+	}
+	if err := CheckPasswordLength(password); err != nil {
```
Comment (unknwon) [inline internal/database/users.go:989]: Why can't we just rely on `CheckPasswordLength` to do the empty check? (which has min requirement) 🤔

**Review by unknwon — APPROVED** [review]

File: internal/database/users.go (hunk 3 above)
Comment (ryanslade) [reply internal/database/users.go:989]: Good point

**Pull request merged**